src/TLuaInterpreter.cpp

```diff
@@ -9009,16 +9009,63 @@ int TLuaInterpreter::createMapLabel(lua_State* L)
 int TLuaInterpreter::setMapZoom(lua_State* L)
 {
     qreal zoom = getVerifiedFloat(L, __func__, 1, "zoom");
+    std::optional<int> areaID;
+    if (lua_gettop(L) > 1) {
+        areaID = getVerifiedInt(L, __func__, 2, "area id", true);
+    }
     Host& host = getHostFromLua(L);
-    if (host.mpMap) {
-        if (host.mpMap->mpMapper) {
-            if (host.mpMap->mpMapper->mp2dMap) {
-                host.mpMap->mpMapper->mp2dMap->setMapZoom(zoom);
-                updateMap(L);
-            }
+    if (host.mpMap.isNull() || host.mpMap->mpMapper.isNull()) {
+        return warnArgumentValue(L, __func__, "no map loaded or no active mapper");
+    }
+
+    if (zoom < T2DMap::csmMinXYZoom) {
+        return warnArgumentValue(L, __func__, qsl("%1 is less than the minimum value allowed for zoom which is %2").arg(QString::number(zoom), QString::number(T2DMap::csmMinXYZoom)));
```

Comment:
Align language to other warnings in this file.
```suggestion
        return warnArgumentValue(L, __func__, qsl("zoom %1 is invalid, it must be greater than %2").arg(QString::number(zoom), QString::number(T2DMap::csmMinXYZoom)));
``` labelID %1 is

Reply:
Technically it should be "greater than or equal to" or "not less than" - the second is shorter in English, not sure if that negative form is the clearest to use though.

Reply:
Fair enough.
```suggestion
        return warnArgumentValue(L, __func__, qsl("zoom %1 is invalid, it must be at least %2").arg(QString::number(zoom), QString::number(T2DMap::csmMinXYZoom)));
```
You're welcome.